autogpt/core/command/base.py

```diff
@@ -1,9 +1,124 @@
 import abc
+import importlib
+import inspect
+from typing import Any, Callable, Optional, Tuple, Dict, Boolean
+from dataclasses import dataclass
 
+@dataclass
+class CommandResult:
+    ok: Boolean
+    message: str
 
 class Command(abc.ABC):
-    pass
+    """A class representing a command. Commands are actions which an agent can take.
+
+	Attributes:
+		name (str): The name of the command.
+		description (str): A brief description of what the command does.
+		signature (str): The signature of the function that theicommand executes. Defaults to None.
+	"""
+    def __init__(
+		self,
+		name: str,
+		description: str,
+		method: Callable[..., Any],
+		signature: str = "",
+		enabled: bool = True,
+		disabled_reason: Optional[str] = None,
+	):
+        self.name = name
+        self.description = description
+        self.method = method
+        self.signature = signature if signature else str(inspect.signature(self.method))
+        self.enabled = enabled
+        self.disabled_reason = disabled_reason
+
+    def __call__(self, *args, **kwargs) -> CommandResult:
+        if not self.enabled:
+            return f"Command '{self.name}' is disabled: {self.disabled_reason}"
```

Comment:
The string concatenation could be replaced with f-strings for improved readability and performance. This also makes the code more consistent in terms of string formatting.

e.g. 
This: `return f"Command '{self.name}' is disabled: {self.disabled_reason}"`

To: `return CommandResult(False, f"Command '{self.name}' is disabled: {self.disabled_reason}")`